feature_engine/timeseries/forecasting/time_lag.py

```diff
@@ -0,0 +1,188 @@
+# Authors: Morgan Sell <morganpsell@gmail.com>
+# License: BSD 3 clause
+
+from typing import List, Union
+
+import pandas as pd
+from pandas.tseries.offsets import DateOffset
+from pandas.tseries.frequencies import to_offset
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from datetime import datetime
+
+from feature_engine.dataframe_checks import (
+    _is_dataframe,
+)
+from feature_engine.variable_manipulation import (
+    _find_all_variables,
+    _find_or_check_numerical_variables,
+    _check_input_parameter_variables,
+)
+
+
+class TimeSeriesLagTransformer(BaseEstimator, TransformerMixin):
+    """
+
+    Parameters
+    ----------
+    periods: int, default=1
+        Number of periods to shift. Can be positive or negative.
+
+    freq: str, default=None
+        Optional increment to use from the tseries module or time rule.
+
+    axis: int, default=1
+        Shift direction. Index is '0'. Columns are '1'.
+
+    fill_value: [int, float, str], default=None
+        Character to be used to fill missing values.
+
+    drop_original: bool, default=True
+        Determines whether the dataframe keeps the original columns
+        that are transformed.
+
+    Attributes
+    ----------
+
+    variables_numerical_:
+        The group of numerical input variables.
+
+    n_features_in_:
+        The number of features in the dataframe.
+
+    Methods
+    -------
+    fit:
+        Obtain attributes and determine whether the dataframe is eligible to
+        be transformed.
+
+    transform:
+        Perform the time-lag transformation on the provided dataframe.
+
+    rename_variables:
+        Create names for the new time-lagged variables.
+
+    Notes
+    -----
+
+
+    See Also
+    --------
+
+
+    References
+    ----------
+
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        periods: int = 1,
+        freq: str = None,
+        fill_value: [int, str] = None,
+        drop_original: bool = False,
+
+    ) -> None:
+
+        if not isinstance(drop_original, bool):
+            raise ValueError(
+                f"'drop_original' is {drop_original}. The variable must "
+                f"be boolean."
+            )
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.periods = periods
+        self.freq = freq
+        self.fill_value = fill_value
+        self.drop_original = drop_original
+
+    def fit(self, X: pd.DataFrame) -> None:
+        """
+        Creates lag
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+           The dataframe containing the variables that that will be lagged.
+
+        Returns
+        -------
+
+        """
+        # check if 'X' is a dataframe
+        _is_dataframe(X)
+
+        # check variables
+        self.variables_ = _find_all_variables(X, self.variables)
```

Comment:
instead of _find_all_variables, we should use _find_or_check_numerical_variables. This transformer only works with numerical variables.